starknet_py/tests/e2e/account/account_test.py

```diff
@@ -25,10 +26,16 @@
 
 @pytest.mark.run_on_devnet
 @pytest.mark.asyncio
-async def test_get_balance_throws_when_token_not_specified(account):
+@pytest.mark.parametrize("client_class", [GatewayClient, FullNodeClient])
+async def test_get_balance_throws_when_token_not_specified(account, client_class):
+    client = (
+        GatewayClient(net="custom.net")
+        if client_class is GatewayClient
+        else FullNodeClient(node_url="custom.net/rpc")
+    )
```

Comment:
You're not really using the class reference from `client_class` beyond creating the client, so I'd either change it to 
```py
client = (
        client_class(net="custom.net")
        if client_class is GatewayClient
        else client_class(node_url="custom.net/rpc")
    )
```

or modify the parametrize entirely

```py
@pytest.mark.parametrize("client_class", ["GatewayClient, "FullNodeClient"])
async def test_get_balance_throws_when_token_not_specified(account, client_class):
    client = (
        GatewayClient(net="custom.net")
        if client_class == "GatewayClient"
        else FullNodeClient(node_url="custom.net/rpc")
    )
```

Not sure which approach is better, but I'd choose one for consistency.